osbrain/tests/common.py

```diff
@@ -87,11 +87,49 @@ def agent_dies(agent, nsproxy, timeout=1.):
     return True
 
 
-def wait_agent_list(agent, name='received', length=None, data=None,
+def attribute_match(attribute, length=None, data=None, value=None):
+    """
+    Check if an attribute matches any condition:
+
+    - Minimum length.
+    - Contains an item.
+    - Is exactly a value.
+
+    Parameters
+    ----------
+    attribute : anything
+        The attribute to match against.
+    length : int, default is None
+        Return True if the attribute has this minimum lenght.
+    data : anything, default is None
+        Return True if the attribute contains this value.
+    value : anything, default is None
+        Return True if the attribute contains this value.
+
+    Returns
+    -------
+    bool
+        Whether the attribute matches any condition.
+    """
+    assert length is not None or data is not None or value is not None, \
+        'No condition passed! will return False always...'
+    if value is not None and attribute == value:
+        return True
+    if length is not None and len(attribute) >= length:
+        return True
+    if data is not None and data in attribute:
+        return True
```

Comment:
As a nitpicky observation, both in the function parameters and the docstrings, order is `length, data, value`. Could we use the same order here? ^^